File: backend_core_mobile/app/models/credito_model.py
# credito_model.py
import uuid
from sqlalchemy import Column, String, Integer, DateTime, ForeignKey, Numeric, Date
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.database.session import Base

class Credito(Base):
    __tablename__ = "cr_creditos"

    id_credito = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    id_solicitud = Column(UUID(as_uuid=True), ForeignKey("solicitudes_credito.id_solicitud"), nullable=True)
    id_cliente = Column(UUID(as_uuid=True), ForeignKey("clientes.id_cliente"), nullable=False)
    numero_credito = Column(String(30), unique=True, nullable=False)
    producto = Column(String(120), nullable=False)
    monto_desembolsado = Column(Numeric(12, 2), nullable=False)
    saldo_capital = Column(Numeric(12, 2), nullable=False)
    plazo_meses = Column(Integer, nullable=False)
    tea = Column(Numeric(5, 2), nullable=False)
    tem = Column(Numeric(8, 6), nullable=False)
    cuota_mensual = Column(Numeric(12, 2), nullable=False)
    fecha_desembolso = Column(Date, nullable=False)
    dia_pago = Column(Integer, nullable=False)
    estado = Column(String(30), nullable=False) # VIGENTE, CANCELADO, VENCIDO
    created_at = Column(DateTime(timezone=True))
    updated_at = Column(DateTime(timezone=True))

    cliente = relationship("Cliente", foreign_keys=[id_cliente])
    solicitud = relationship("SolicitudCredito", foreign_keys=[id_solicitud])

File: mobile_app_bcp/lib/shared/models/credito_model.dart
// shared/models/credito_model.dart
class CreditoModel {
  final String idCredito;
  final String idSolicitud;
  final String idCliente;
  final String numeroCredito;
  final String? producto;
  final double montoDesembolsado;
  final double saldoCapital;
  final int plazoMeses;
  final double tea;
  final double tem;
  final double cuotaMensual;
  final DateTime? fechaDesembolso;
  final int? diaPago;
  final String estado;

  CreditoModel({
    required this.idCredito,
    required this.idSolicitud,
    required this.idCliente,
    required this.numeroCredito,
    this.producto,
    this.montoDesembolsado = 0,
    this.saldoCapital = 0,
    this.plazoMeses = 0,
    this.tea = 0,
    this.tem = 0,
    this.cuotaMensual = 0,
    this.fechaDesembolso,
    this.diaPago,
    this.estado = 'VIGENTE',
  });

  String get montoFormateado => 'S/ ${montoDesembolsado.toStringAsFixed(2)}';
  String get saldoFormateado => 'S/ ${saldoCapital.toStringAsFixed(2)}';
  String get cuotaFormateada => 'S/ ${cuotaMensual.toStringAsFixed(2)}';

  factory CreditoModel.fromJson(Map<String, dynamic> json) {
    return CreditoModel(
      idCredito: json['id_credito'] ?? '',
      idSolicitud: json['id_solicitud'] ?? '',
      idCliente: json['id_cliente'] ?? '',
      numeroCredito: json['numero_credito'] ?? '',
      producto: json['producto'],
      montoDesembolsado: (json['monto_desembolsado'] ?? 0).toDouble(),
      saldoCapital: (json['saldo_capital'] ?? 0).toDouble(),
      plazoMeses: json['plazo_meses'] ?? 0,
      tea: (json['tea'] ?? 0).toDouble(),
      tem: (json['tem'] ?? 0).toDouble(),
      cuotaMensual: (json['cuota_mensual'] ?? 0).toDouble(),
      fechaDesembolso: json['fecha_desembolso'] != null
          ? DateTime.tryParse(json['fecha_desembolso'])
          : null,
      diaPago: json['dia_pago'],
      estado: json['estado'] ?? 'VIGENTE',
    );
  }
}
